etl/steps/data/garden/war/2023-07-21/prio_v31.py

```diff
@@ -0,0 +1,294 @@
+
+    ## Rename
+    tb_ongoing = tb_ongoing.rename(  # type: ignore
+        columns={
+            "id": "number_ongoing_conflicts",
+            "bdeadlow": "number_deaths_ongoing_conflicts_battle_low",
+            "bdeadhig": "number_deaths_ongoing_conflicts_battle_high",
+            "bdeadbes": "number_deaths_ongoing_conflicts_battle",
+        }
+    )
+
+    return tb_ongoing
+
+
+def _add_new_metrics(tb: Table) -> Table:
+    # Reduce table
+    tb_new = tb.sort_values("year").drop_duplicates(subset=["id", "region"], keep="first")[
+        ["year", "region", "type", "id"]
+    ]
+    assert (
+        tb_new["id"].value_counts().max() == 1
+    ), "There are multiple instances of a conflict with the same ID. Maybe same conflict in different regions or with different types? This is assumed not to happen"
+
+    # Estimate metric for regions and types
+    tb_new_regions = tb_new.groupby(["year", "region", "type"], as_index=False)["id"].nunique()
+
+    # Estimate metric for new type='all'
+    tb_new_alltype = tb_new_regions.groupby(["year", "region"], as_index=False)["id"].sum()
+    tb_new_alltype["type"] = "all"
+
+    # Estimate metric for new type='intrastate'
+    tb_new_regions_intra_ = tb_new_regions[tb_new_regions["type"].isin([3, 4])]
+    tb_new_intratype = tb_new_regions_intra_.groupby(["year", "region"], as_index=False)["id"].sum()
+    tb_new_intratype["type"] = "intrastate"
+
+    tb_new = pd.concat([tb_new_regions, tb_new_alltype, tb_new_intratype], ignore_index=True)
+
+    # Estimate metric for new region='World'
+    tb_new_world = tb_new.groupby(["year", "type"], as_index=False)["id"].sum()
+    tb_new_world["region"] = "World"
+
+    # Combine
+    tb_new = pd.concat([tb_new, tb_new_world], ignore_index=True)
+
+    # Rename
+    tb_new = tb_new.rename(columns={"id": "number_new_conflicts"})  # type: ignore
+
+    return tb_new
+
+
+def replace_missing_data_with_zeros(tb: Table) -> Table:
+    """Replace missing data with zeros.
+
+    In some instances there is missing data. Instead, we'd like this to be zero-valued.
+    """
+    # Add missing (year, region, conflict_typ) entries (filled with NaNs)
+    years = np.arange(tb["year"].min(), tb["year"].max() + 1)
+    regions = set(tb["region"])
+    conflict_types = set(tb["conflict_type"])
+    new_idx = pd.MultiIndex.from_product([years, regions, conflict_types], names=["year", "region", "conflict_type"])
+    tb = tb.set_index(["year", "region", "conflict_type"]).reindex(new_idx).reset_index()
+
+    # Change NaNs for 0 for specific rows
+    ## For columns "number_ongoing_conflicts", "number_new_conflicts"
+    columns = [
+        "number_ongoing_conflicts",
+        "number_new_conflicts",
+        "number_deaths_ongoing_conflicts_battle_high",
+        "number_deaths_ongoing_conflicts_battle_low",
+    ]
+    tb.loc[:, columns] = tb.loc[:, columns].fillna(0)
+
+    # Set number of deaths to zero whenever high and low estimates are zero
```

Comment:
What are the problems with estimating a value for `number_deaths_ongoing_conflicts_battle` when it's missing and high and low are not? Just as the previous dataset, I wonder how you'd solve that on the explorers. Probably showing the three together for all the views

Reply:
`number_deaths_ongoing_conflicts_battle` is missing for some rows in the source; hence we can't obtain it.

In these lines, we fill NaNs in `number_deaths_ongoing_conflicts_battle` only when values in high and low estimates are also zero.

Reply:
I know, it just came to my mind the question here